cmd/asset-syncer/utils.go

```diff
@@ -124,23 +124,31 @@ func getSha256(src []byte) (string, error) {
 
 // Repo defines the methods to retrive information from the given repository
 type Repo interface {
-	Checksum() (string, error)
+	Index() (interface{}, error)
+	Checksum(interface{}) (string, error)
 	Repo() *models.RepoInternal
-	Charts() ([]models.Chart, error)
+	Charts(interface{}) ([]models.Chart, error)
 	FetchFiles(name string, cv models.ChartVersion) (map[string]string, error)
 }
 
 // HelmRepo implements the Repo interface for chartmuseum-like repositories
 type HelmRepo struct {
-	content []byte
```

Comment:
I don't think we'd need to get rid of this state - it makes sense to only fetch the index once, even if we include an `Index()` on the interface, it be memoized (only fetch if not already present).